modules/add_credit/src/AddCreditService.php

```diff
@@ -117,6 +117,19 @@ public function entityBaseFieldInfo(EntityTypeInterface $entity_type) {
         ->setTranslatable(TRUE)
         ->setDefaultValue(FALSE);
 
+      // Add base fields to custom amount.
+      $fields['apigee_add_credit_custom_amount'] = BaseFieldDefinition::create('boolean')
+        ->setLabel(t('Allow user to enter a custom amount'))
+        ->setRevisionable(TRUE)
+        ->setTranslatable(TRUE)
+        ->setDefaultValue(FALSE);
+
+      $fields['apigee_add_credit_minimum_amount'] = BaseFieldDefinition::create('commerce_price')
```

Comment:
For a typical product, the pricing is handled in the variation. Can we move all of this to the variation or create a new variation type that handles this?

<img width="794" alt="image" src="https://user-images.githubusercontent.com/159693/51012492-3e47d380-1512-11e9-86c0-d445736c7f36.png">